implementations/rust/ockam/ockam_command/src/message/send.rs

```diff
@@ -91,13 +95,40 @@ async fn rpc(mut ctx: Context, (opts, cmd): (CommandGlobalOpts, SendCommand)) ->
         let mut rpc = RpcBuilder::new(ctx, opts, &api_node)
             .tcp(tcp.as_ref())?
             .build();
-        rpc.request_with_timeout(req(&to, &cmd.message), Duration::from_secs(cmd.timeout))
+
+        // must bind hex decoding vec to later get its slice
+        let hex_decode_result = hex::decode(cmd.message.to_owned());
+        let hex_bytes = match hex_decode_result {
+            Ok(val) => {
+                val
+            },
+            Err(err) => {
+                if cmd.hex {
+                    // only raise hex decoding error if hex flag is set
+                    return Err(err.into())
+                }
+                // return empty vec that will never be used, to satisfy match
+                vec![]
+            }
+        };
+
+        let msg_req = if cmd.hex {
+            req(&to, hex_bytes.as_slice())
+        } else {
+            req(&to, cmd.message.as_bytes())
+        };
```

Comment:
This is close to what I was suggesting in my previous comment but note how you have the `if cmd.hex` intertwined across all the logic here.

The trick here is to return the appropriate types at each step to avoid the problems you mentioned. This is how I'd do it so that we only use the `hex` flag to encode/decode the message:

```rust
// Prepare request
let mut rpc = RpcBuilder::new(ctx, opts, &api_node)
    .tcp(tcp.as_ref())?
    .build();
let to = crate::project::util::clean_projects_multiaddr(to, projects_sc)?;
let message = if cmd.hex {
    hex::decode(cmd.message)?
} else {
    cmd.message.as_bytes().to_vec()
};
let req = req(&to, message.as_slice());

// Send request
rpc.request_with_timeout(req, Duration::from_secs(cmd.timeout))
    .await?;
let res = {
    let res = rpc.parse_response::<Vec<u8>>()?;
    if cmd.hex {
        hex::encode(res)
    } else {
        String::from_utf8(res).context("Received content is not a valid utf8 string")?
    }
};
```